sdk/lib/crt/ucrtsupport.cmake

```diff
@@ -2,28 +2,80 @@
 list(APPEND UCRTSUPP_SOURCE
     ${CRT_FLOAT_SOURCE}
     ${CRT_MATH_SOURCE}
+    misc/amsg.c
+    misc/purecall.c
+    misc/tls.c
+    wine/cpp.c
+    wine/except.c
     wine/undname.c
 )
 
 list(APPEND UCRTSUPP_ASM_SOURCE
-    ${CRT_EXCEPT_ASM_SOURCE}
     ${CRT_FLOAT_ASM_SOURCE}
     ${CRT_MATH_ASM_SOURCE}
     ${CRT_SETJMP_ASM_SOURCE}
+    ${CRT_WINE_ASM_SOURCE}
 )
 
+if(ARCH STREQUAL "i386")
+    list(APPEND UCRTSUPP_SOURCE
+        except/i386/CxxHandleV8Frame.c
+        wine/except_i386.c
+    )
+    list(APPEND UCRTSUPP_ASM_SOURCE
+        except/i386/__CxxFrameHandler3.s
+        except/i386/chkesp.s
+        wine/rosglue_i386.s
+    )
+    if(MSVC)
+        list(APPEND UCRTSUPP_ASM_SOURCE
+            except/i386/cpp.s
+            except/i386/prolog.s
+        )
+    endif()
+elseif(ARCH STREQUAL "amd64")
+    list(APPEND UCRTSUPP_SOURCE
+        wine/except_x86_64.c
+    )
+    if(MSVC)
+        list(APPEND UCRTSUPP_ASM_SOURCE
+            except/amd64/cpp.s
+        )
+    endif()
+elseif(ARCH STREQUAL "arm")
+    list(APPEND UCRTSUPP_SOURCE
+        wine/except_arm.c
+    )
+    if(MSVC)
+        list(APPEND UCRTSUPP_ASM_SOURCE
+            except/arm/cpp.s
+        )
+    endif()
+elseif(ARCH STREQUAL "arm64")
+    list(APPEND UCRTSUPP_SOURCE
+        wine/except_arm64.c
+    )
+endif()
+
+#set_source_files_properties(${UCRTSUPP_ASM_SOURCE} PROPERTIES COMPILE_DEFINITIONS "__MINGW_IMPORT=extern;USE_MSVCRT_PREFIX;_MSVCRT_LIB_;_MSVCRT_;_MT;CRTDLL")
```

Comment:
Is this going to be useful in the future?

Reply:
No, that's obsolete, will remove.